shimmingtoolbox/masking/shape.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+"""
+Image mask with shape API
+"""
+
+import numpy as np
+
+
+def shape(data, shape, center_x=None, center_y=None, center_z=None, len_x=None, len_y=None, len_z=None, radius=None):
+    """
+    Wrapper to different shape masking algos
+
+    Args:
+
+    Returns:
+        nd.array: mask
+    """
+
+    # allowed_shapes = {'square', 'cube', 'disk', 'ball'}
+
+    # Make sure input params are valid
+    if data.ndim not in [2, 3]:
+        raise RuntimeError('Number of dimensions not supported')
+
+    # TODO: Make sure specified center and length fit within data.shape
+
+    # Create mask with all zeros
+    mask = np.zeros_like(data)
+
+    if shape == 'square':
+        mask = shape_square(data, center_x, center_y, len_x, len_y)
+    elif shape == 'cube':
+        # TODO
+        pass
+    elif shape == 'disk':
+        # TODO
+        pass
+    elif shape == 'ball':
+        # TODO
+        pass
+    else:
+        raise RuntimeError('Input shapes not part of the allowed shapes for masking')
```

Comment:
A good way to avoid using if elif is to make a global dictionary with all the functions for the shapes keyed with the shape name and only call this function:
```suggestion
   shape_mask[shape](data, mask_info)
```
And at the beginning of the document:
```
shape_mask = {'square': shape_square, 'cube': shape_cube, 'disk': shape_disk, 'ball': shape_ball}
```
This way, it is easier to call. You can also unwrap the information in the mask_info to use with you're current function call.

Reply:
That's an awesome tip, thanks. I had to modify the `shape_mask` function call to `shape_mask[shape](data, **mask_info)` to keep the original function declaration.